### PR TITLE
Add ghostify function

### PR DESCRIPTION
**Pull request by antalsz**

### CODE REVIEW

File: ocaml/parsing/parser.mly [hunk 1]
```diff
@@ -504,7 +504,7 @@ let lapply ~loc p1 p2 =
 let loc_map (f : 'a -> 'b) (x : 'a Location.loc) : 'b Location.loc =
   { x with txt = f x.txt }
 
-let make_ghost x = { x with loc = { x.loc with loc_ghost = true }}
+let make_ghost x = { x with loc = Location.ghostify x.loc }
```
Comment (goldfirere) [inline ocaml/parsing/parser.mly:507]: This could avoid allocation by testing for the ghost before proceeding.

**Review by goldfirere — COMMENTED** [review]
The rest of the ghostify commit looks good, but this PR contains a bunch of other stuff, so it's not mergeable as-is. If possible, it's best to have a PR tackle just one problem, and then it can be merged separately. Stacking PRs works well enough in practice, in my experience, where one PR targets another.

**Comment by antalsz** [comment]
Whoops, I gave this PR the wrong base, I'll recreate it

**Comment by antalsz** [comment]
This is now antalsz#1

File: ocaml/parsing/parser.mly (hunk 1 above)
Comment (antalsz) [reply ocaml/parsing/parser.mly:507]: Addressed in antalsz#1

**Pull request closed without merging**